src/smartcontracts/smartContract.ts

```diff
@@ -135,15 +136,12 @@ export class SmartContract implements ISmartContract {
             isPayableBySmartContract: metadataAsJson.payableBySc
         });
 
-        return new Transaction({
-            receiver: Address.fromBech32(intent.receiver),
-            sender: Address.fromBech32(intent.sender),
-            value: value,
-            gasLimit: new BigNumber(intent.gasLimit).toNumber(),
-            gasPrice: gasPrice,
-            data: new TransactionPayload(Buffer.from(intent.data!)),
-            chainID: chainID
-        });
+        let transaction = Transaction.fromDraft(draftTx);
```

Comment:
Can also be `const` (even if we mutate its inner state).

Reply:
It's now `const`.